fw.lua

```diff
@@ -442,6 +483,7 @@ local function _rule_action(self, action, ctx, collections)
 		ACCEPT = function(self, ctx)
 			_log(self, "An explicit ACCEPT was sent, so ending this phase with ngx.OK")
 			if (self._mode == "ACTIVE") then
+				_log_request(self, ctx, collections)
```

Comment:
Recent work on the development branch contains a few concepts missing here:

- a _finalize() function that seems like an appropriate place to put _log_request()
- the concept of 'altering' actions (e.g. actions that affect the transaction by calling ngx.exit(); this also ties in with _finalize()